### DISCUSSION TITLE
useQuery on background without causing re-render of component

### DISCUSSION THREAD

**Discussion by lironezra** [TanStack/query#3351] [Q&A]
Hi guys,

Suppose I have to perform an API fetching and I want to do that with `useQuery`, but the problem is that it's causing the component to re-render again.
there is an option to use `useQuery` but without this re-render?

I'll give an example, after all the business logic ended I need to perform request to the server to update something in the DB, after that action was finished I don't need to do anything in the component and the component shouldn't re-render, but if I am using `useQuery `it will.

Thanks all!

**Comment by TkDodo** [comment]
For updating something, you should `useMutation()`
